### PR TITLE
minor change to fe_tools_interpolate to allow complex-valued vectors

### PR DESCRIPTION
**Pull request by davydden**
in reference to https://github.com/dealii/dealii/issues/2033

### CODE REVIEW

File: source/fe/fe_tools_interpolate.cc [hunk 1]
```diff
@@ -222,7 +222,8 @@ namespace FETools
     for (types::global_dof_index i=0; i<dof2.n_dofs(); ++i)
       if (locally_owned_dofs.is_element(i))
         {
-          Assert(touch_count(i) != typename OutVector::value_type(),
+          const typename OutVector::value_type touch_i = touch_count(i);
+          Assert(touch_i != typename OutVector::value_type(0),
```
Comment (bangerth) [inline source/fe/fe_tools_interpolate.cc]: You'll get a warning about an unused variable here in release mode. Please just do the cast by hand inside the assertion.

Comment (davydden) [inline source/fe/fe_tools_interpolate.cc]: i was not able to get it work inside an Assert, how would you write it?

Comment (bangerth) [inline source/fe/fe_tools_interpolate.cc]: I think this should do:
```
  Assert (static_cast<typename OutVector::value_type>(touch_count(i)) != typename OutVector::value_type(), ...);
```

Comment (davydden) [inline source/fe/fe_tools_interpolate.cc]: thanks, i think i missed `static_cast` last time i tried.

**Pull request merged**